### ISSUE TITLE
USB3 to sata adapter with 250GB HDD and firmware question

### ISSUE THREAD

**Issue by flyoffthehandle** [Chadster766/McDebian#34]
1. Has anyone run into issues with previous firmware booting after trying to update using the mcdebian image file on a wrt1900acv1?

2. I've tried the above many times with both usb drives and a HDD using a sata to usb3adapter that had rootfs copied to them. Is the later even supported?

Thanks for any help.

**Comment by Chadster766** [comment]
You need to update the McDebian rootfs to match the firmware file you are using.

https://github.com/Chadster766/McDebian/wiki/X.-McDebian-Updates

**Comment by Chadster766** [comment]
Yes for the WRT1900ACS you upgrade from a running McDebian 4.9.12 directly to 4.9.31.

**Comment by flyoffthehandle** [comment]
So for the initial switch from linksys or dd-wrt, which is what I'm trying, I've followed the wiki and used the 4.4.16 firmware/4.4.16 rootfs files. I formatted USB ext4 after wiping the partition table and creating a new one with 1 partition. Mounted it and uncompressed the rootfs tar file into the mounted directory. (after waiting for a sync to finish I made sure to do a few syncs for good measure) I unmounted the USB, plugged in to the back of the linksys (I've tried flashing firmware while usb was in usb slot 1 and 2 and have made sure the linksys was mounting the usb drive before starting the firmware upgrade). Go to firmware upgrade page of router and then initiate the firmware upgrade using the file from the wiki. After restart I dd-wrt loads instead of the debian system. I've used ssh/webUI command page to manually restart the router and dd-wrt loads each time.

If all of that looks like the proper steps you take to get mcdebian running on the router then I believe I'm out of ideas. Sometime I might nuke the dd-wrt and go back to the stock firmware to see if it's the firmware that is causing this from working.

In your comment you say "for the WRT1900ACS" I have been trying this with a WRT1900AC (version 1).

firmware being used to flash:
http://www.protechs-online.com/downloads/McDebian/firmwares/McDebian-WRT1900AC-V1-FW_VER1_kernel_4_4_16.img

rootfs that has been used with both a USB stick and also tried with a hard drive connected via USB3toSATA adapter:
http://www.protechs-online.com/downloads/McDebian/rootfs/mcdebian-router-WRT-1900-1200-Kernel_4_4_16-base.gz

Boots DD-WRT firmware no matter what I try or USB port I have the rootfs plugged in to when it restarts after the FW update finishes.

**Comment by Chadster766** [comment]
Sorry I got your router model wrong at first but all your steps are good for the WRT1900AC V1.

Try manually switching firmware load locations by turning off and on the router three times in a row right when the Ethernet LEDs' start flashing.

**Comment by flyoffthehandle** [comment]
Updating this issue-

As I'm unfamiliar regarding specifically why DD-WRT was doing a "dummy" firmware install process I will just document the version here and obviously what ended up working.

Running firmware DD-WRT v3.0-r30015M kongmv (06/24/16) on router was failing to upgrade using the initial mcdebian FW file but didn't show any errors and adding to my confusion the fact the router would reboot after showing the firmware had loaded properly, as they would do after a firmware update. 
It would come back up still running DD-WRT (I tried with "reset after upgrade"  enabled, and without).

Solution was, as you suggested, to go through the same "known to work" process with the router running stock linksys firmware (this firmware was upgraded to the most current release before mcdebian installation attempt). After reboot, I waited about 30 minutes and then did one quick power cycle of the router using the switch on the back.

Mcdebian loaded successfully.
